internal/ethapi/transaction_args.go

```diff
@@ -72,12 +74,31 @@ func (arg *TransactionArgs) data() []byte {
 
 // setDefaults fills in default values for unspecified tx fields.
 func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
-	if args.GasPrice == nil {
-		price, err := b.SuggestPrice(ctx)
-		if err != nil {
-			return err
+	if b.ChainConfig().IsLondon(b.CurrentBlock().Number()) && args.GasPrice == nil {
```

Comment:
Seems weird that we check gas price for nil-ness and then set tip and feecap. Nowhere did we check if this is a 1559 transaction or a plain one.

Reply:
I guess we can default to 1559 if neither gasPrice, feeCap or tip is set.

I think we need to check properly various possible combinations, reject most of them (price + feecap, price + tipcap)

Reply:
Something along the lines of

```
if args.GasPrice != nil && (args.FeeCap != nil || args.Tip != nil) { return error }
```

The rest should be ok after this check.